domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -456,25 +429,33 @@ class TopicListController @Inject constructor(
     )
   }
 
+  private fun getDependentLisOfTopicForOngoingTopic(topicId: String): List<String> {
+    val listOfTopicIds = mutableListOf<String>()
+    if (topicId == TEST_TOPIC_ID_0) {
+      listOfTopicIds.add(FRACTIONS_TOPIC_ID)
+    } else if (topicId == TEST_TOPIC_ID_1) {
+      listOfTopicIds.add(TEST_TOPIC_ID_0)
+      listOfTopicIds.add(RATIOS_TOPIC_ID)
+    }
+    return listOfTopicIds
+  }
+
   private fun computeSuggestedStories(
     topicProgressList: List<TopicProgress>
   ): List<PromotedStory> {
     val recommendedStories = mutableListOf<PromotedStory>()
-
-    val topicIdJsonArray = jsonAssetRetriever
-      .loadJsonFromAsset("topics.json")!!
-      .getJSONArray("topic_id_list")
-
-    val topicIdList = (0 until topicIdJsonArray.length()).map { topicIdJsonArray[it].toString() }
-
-    val index = topicIdList.indexOf(topicProgressList.last().topicId)
-
-    for (i in (index + 1) until topicIdJsonArray.length()) {
-      if (topicIdJsonArray.length() > i) {
-        val recommendedStoriesIdFromAssets =
-          createRecommendedStoryFromAssets(topicIdJsonArray[i].toString())
-        if (recommendedStoriesIdFromAssets != null)
-          recommendedStories.add(recommendedStoriesIdFromAssets)
+    topicProgressList.forEach {
+      val dependentListOfTopics =
+        getDependentLisOfTopicForOngoingTopic(it.topicId)
+      for (j in dependentListOfTopics.indices) {
+        val found = topicProgressList.any { it.topicId == dependentListOfTopics[j] }
```

Comment:
Isn't the algorithm that we need: create a list of all learned topic IDs then iterate through all topic IDs and, for each one, check that topic's list of dependencies against the list of learned topic IDs? If a topic's list of dependent IDs is fully contained in the list of learned IDs then it qualifies for being recommended.

Reply:
Done.